### PR TITLE
 Update Self Host to explain --detach and updating

### PR DESCRIPTION
**Pull request by dm-canteen**
~~Use -d on all `docker compose up` commands, and include a reminder to restart the containers after pulling the latest images.~~

Related Discord message: https://discord.com/channels/930051556043276338/930051949888405524/1140473986053054474

### CODE REVIEW

**Comment by rubenfiszel** [comment]
I agree one should use -d but my 2 cents is that to debug/iterate, you do not want to use it in docker mode. 

Your update section is using more of a special behavior which is that `docker compose up -d` will restart the containers if already in deamon mode.

So I would rephrase something to this effect:

```
In practice, you want to run the docker compose in deamon mode with the `-d` parameter as follows:
xxx
```

and a little tip under

```
:::tip

to pull latest, it is sufficient to apply `docker compose up -d` if your docker is already in deamon mode, since it will restart the containers and they are on pull latest policy.
```

Thanks for the contribution!

**Comment by dm-canteen** [comment]
Agreed, I did not know you could run `up -d` again without pulling, thanks for the tip!
Let me know if there's any issues now.
